sympy/solvers/ode/tests/test_systems.py

```diff
@@ -0,0 +1,457 @@
+            Eq(y(t), C1*exp(2*t)*cos(t) - C2*exp(2*t)*sin(t))]
+   assert dsolve(eq11) == sol11
+   assert checksysodesol(eq11, sol11) == (True, [0, 0])
+
+   eq12 = [Eq(x(t).diff(t), x(t) + 2*y(t)), Eq(y(t).diff(t), 2*x(t) + y(t))]
+   #sol12 = [Eq(x(t), C1*exp(-t) + C2*exp(3*t)),
+   #         Eq(y(t), -C1*exp(-t) + C2*exp(3*t))]
+   sol12 = [Eq(x(t), -C1*exp(-t) + C2*exp(3*t)), Eq(y(t), C1*exp(-t) + C2*exp(3*t))]
+   assert dsolve(eq12) == sol12
+   assert checksysodesol(eq12, sol12) == (True, [0, 0])
+
+   eq13 = [Eq(x(t).diff(t), 4*x(t) + y(t)), Eq(y(t).diff(t), -x(t) + 2*y(t))]
+   sol13 = [Eq(x(t), C1*exp(3*t) + C2*(t*exp(3*t) + exp(3*t))),
+            Eq(y(t), -C1*exp(3*t) - C2*t*exp(3*t))]
+   assert dsolve(eq13) == sol13
+   assert checksysodesol(eq13, sol13) == (True, [0, 0])
+
+   eq14 = [Eq(x(t).diff(t), a*y(t)), Eq(y(t).diff(t), a*x(t))]
+   sol14 = [Eq(x(t), -C1*exp(-a*t) + C2*exp(a*t)), Eq(y(t), C1*exp(-a*t) + C2*exp(a*t))]
+   assert dsolve(eq14) == sol14
+   assert checksysodesol(eq14, sol14) == (True, [0, 0])
+
+   eq15 = [Eq(x(t).diff(t), a*y(t)), Eq(y(t).diff(t), b*x(t))]
+   sol15 = [Eq(x(t), -C1*a*exp(-t*sqrt(a*b))/sqrt(a*b) + C2*a*exp(t*sqrt(a*b))/sqrt(a*b)),
+            Eq(y(t), C1*exp(-t*sqrt(a*b)) + C2*exp(t*sqrt(a*b)))]
+   assert dsolve(eq15) == sol15
+   assert checksysodesol(eq15, sol15) == (True, [0, 0])
+
+   eq16 = [Eq(x(t).diff(t), a*x(t) + b*y(t)), Eq(y(t).diff(t), c*x(t))]
+   sol16 = [Eq(x(t), -2*C1*b*exp(t*(a/2 - sqrt(a**2 + 4*b*c)/2))/(a + sqrt(a**2 + 4*b*c)) - 2*C2*b*exp(t*(a/2 + sqrt(a**2 + 4*b*c)/2))/(a - sqrt(a**2 + 4*b*c))),
+            Eq(y(t), C1*exp(t*(a/2 - sqrt(a**2 + 4*b*c)/2)) + C2*exp(t*(a/2 + sqrt(a**2 + 4*b*c)/2)))]
+   assert dsolve(eq16) == sol16
+   assert checksysodesol(eq16, sol16) == (True, [0, 0])
+
+   Z0 = Function('Z0')
+   Z1 = Function('Z1')
+   Z2 = Function('Z2')
+   Z3 = Function('Z3')
+
+   k01, k10, k20, k21, k23, k30 = symbols('k01 k10 k20 k21 k23 k30')
+
+   eq1 = (Eq(Derivative(Z0(t), t), -k01*Z0(t) + k10*Z1(t) + k20*Z2(t) + k30*Z3(t)), Eq(Derivative(Z1(t), t),
+         k01*Z0(t) - k10*Z1(t) + k21*Z2(t)), Eq(Derivative(Z2(t), t), -(k20 + k21 + k23)*Z2(t)), Eq(Derivative(Z3(t),
+         t), k23*Z2(t) - k30*Z3(t)))
+
+   sol1 = [Eq(Z0(t), C1*k10/k01 + C2*(-k10 + k30)*exp(-k30*t)/(k01 + k10 - k30) - C3*exp(t*(-k01 - k10)) + C4*(-k10*k20 - k10*k21 + k10*k30 + k20**2 + k20*k21 + k20*k23 - k20*k30 - k23*k30)*exp(t*(-k20 - k21 - k23))/(k23*(-k01 - k10 + k20 + k21 + k23))),
+           Eq(Z1(t), C1 - C2*k01*exp(-k30*t)/(k01 + k10 - k30) + C3*exp(t*(-k01 - k10)) + C4*(-k01*k20 - k01*k21 + k01*k30 + k20*k21 + k21**2 + k21*k23 - k21*k30)*exp(t*(-k20 - k21 - k23))/(k23*(-k01 - k10 + k20 + k21 + k23))),
+           Eq(Z2(t), C4*(-k20 - k21 - k23 + k30)*exp(t*(-k20 - k21 - k23))/k23),
+           Eq(Z3(t), C2*exp(-k30*t) + C4*exp(t*(-k20 - k21 - k23)))]
+
+   assert dsolve(eq1, simplify=False) == sol1
+   # assert checksysodesol(eq1, sol1) == (True, [0, 0, 0])
```

Comment:
Does `checksysodesol` not work here?

Reply:
I think it does. I missed this out. I will update this line.